tests/rados/test_osd_async_recovery.py

```diff
@@ -0,0 +1,105 @@
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Performs the test osd sync tests.
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_object = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_object)
+    ceph_nodes = kw.get("ceph_nodes")
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    target_configs = config["cache_trim_max_skip_pinned"]["configurations"]
+    mon_obj.set_config(section="osd", name="osd_async_recovery_min_cost", value="10")
+    mon_obj.set_config(section="osd", name="debug_osd", value="20")
+
+    # Creating pools and starting the test
+    for entry in target_configs.values():
+        log.debug(
+            f"Creating {entry['pool_type']} pool on the cluster with name {entry['pool_name']}"
+        )
+        if entry.get("pool_type", "replicated") == "erasure":
+            method_should_succeed(
+                rados_object.create_erasure_pool, name=entry["pool_name"], **entry
+            )
+        else:
+            method_should_succeed(
+                rados_object.create_pool,
+                **entry,
+            )
+        rados_object.bench_write(
+            pool_name=entry["pool_name"], rados_write_duration=300, background=True
+        )
+
+        log_lines = [
+            "choose_async_recovery_replicated candidates by cost are:"
+            "choose_async_recovery_replicated"
+        ]
+        osd_count = 0
+        # Restart 3 OSDs
+        for node in ceph_nodes:
+            if node.role == "osd":
+                if osd_count == 3:
+                    break
+                init_time, _ = installer.exec_command(
+                    cmd="sudo date '+%Y-%m-%d %H:%M:%S'"
+                )
+                node_osds = rados_object.collect_osd_daemon_ids(node)
+                # Pick a single OSD from the host OSDs list
+                node_osd = random.sample(node_osds, 1)[0]
+
+                if not rados_object.change_osd_state(action="restart", target=node_osd):
+                    log.error(f" Failed to restart OSD : {node_osd}")
+                    return 1
+                time.sleep(10)
+                end_time, _ = installer.exec_command(
+                    cmd="sudo date '+%Y-%m-%d %H:%M:%S'"
+                )
+
+                if not verify_async_recovery_log(
+                    rados_obj=rados_object,
+                    osd=node_osd,
+                    start_time=init_time,
+                    end_time=end_time,
+                    lines=log_lines,
+                ):
+                    log.error("Could not find the log lines")
+                    return 1
+                log.info("Found the log lines")
+            osd_count = osd_count + 1
+
+
+def verify_async_recovery_log(
```

Comment:
Please add doc strings for the function.

Reply:
Added the doc string.